internal/datastore/postgres/common/pgx.go

```diff
@@ -105,7 +106,52 @@ func ConfigurePGXLogger(connConfig *pgx.ConnConfig) {
 	}
 
 	l := zerologadapter.NewLogger(log.Logger, zerologadapter.WithSubDictionary("pgx"))
-	connConfig.Tracer = &tracelog.TraceLog{Logger: levelMappingFn(l), LogLevel: tracelog.LogLevelInfo}
+	addTracer(connConfig, &tracelog.TraceLog{Logger: levelMappingFn(l), LogLevel: tracelog.LogLevelInfo})
+}
+
+// ConfigureOTELTracer adds OTEL tracing to a pgx.ConnConfig
+func ConfigureOTELTracer(connConfig *pgx.ConnConfig) {
+	addTracer(connConfig, otelpgx.NewTracer(otelpgx.WithTrimSQLInSpanName()))
+}
+
+func addTracer(connConfig *pgx.ConnConfig, tracer pgx.QueryTracer) {
+	composedTracer := addComposedTracer(connConfig)
+	composedTracer.Tracers = append(composedTracer.Tracers, tracer)
+}
+
+func addComposedTracer(connConfig *pgx.ConnConfig) *ComposedTracer {
```

Comment:
can we add a test for this?

Reply:
done in https://github.com/authzed/spicedb/pull/1449/commits/2a8a1d548483011f887edf08e56c25db7781d450